### PR TITLE
Don't destroy fragments when switching tabs

### LINKED ISSUE #5615: BottomTabs on Android shell is currently recreating the Page Handler with each navigation

### LINKED ISSUE #5877: OnAppearing firing twice when using Shell.Current.GoToAsync()
### Description

Hi,

OnAppearing fires twice when trying to navigate to a different Page using `Shell.Current.GoToAsync()`.
Furthermore when navigating to the Page once again OnAppearing does not fire at all.
So during the App Lifetime it only (wrongly) fires twice and then never again.

https://github.com/ioiooi/MAUIOnAppearingTwice

In the linked repository you will find a App with three Pages. Page1, Page2 and Page3. Page2 should never be seen because it navigates to Page3 in its OnAppearing. But after the first "run-through" it will be displayed because OnAppearing won't fire anymore.

### Steps to Reproduce

1. Clone repo https://github.com/ioiooi/MAUIOnAppearingTwice
2. Run on physical Android Device or Emulator with Android Version 11
3. Use the Button on Page1 and navigate to Page2
4. Page2 OnAppearing fires twice and navigates to Page3
5. Use the Button on Page3 to navigate to Page2
6. Page2 OnAppearing does not fire at all and now you're stuck :( 

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

### Did you find any workaround?

I have no Idea why but replacing
`await Shell.Current.GoToAsync($"//{nameof(Page3)}");`
with
`Task.Run(async () => await Shell.Current.GoToAsync($"//{nameof(Page3)}"));`
fixes the issue.

After replacing the line the `async` keyword can be removed from OnAppearing().

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

- Don't recreate the entire ShellSection renderer view when you navigate back to it. Just return the previously created view
- Wire up `ShellFragmentStateAdapter` so that it better keeps track of created fragments

### Issues Fixed

Fixes #5615
Fixes #5877

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs [hunk 1]
```diff
@@ -24,26 +28,49 @@ internal class ShellFragmentStateAdapter : FragmentStateAdapter
...
 		{
-			return SectionController.GetItems()[position].GetHashCode();
+			var shellContent = _items[position];
+			foreach (var item in _createdShellContent)
+				if (item.Value == shellContent)
+					return item.Key;
+
+			var id = _id;
+			_createdShellContent.Add(_id++, shellContent);
+			return id;
```
Comment (mattleibow) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs:68]: Does the position not work as the ID?

Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs:68]: I originally had position but I was concerned about that causing an issue if the items get resorted or if someone were to insert something into a new position on the list. It just felt safer for each `item` to have the same unique id until the end of time

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs [hunk 2]
```diff
@@ -57,6 +84,7 @@ protected override void Dispose(bool disposing)
 			{
 				SectionController.ItemsCollectionChanged -= OnItemsCollectionChanged;
 				_shellSection = null;
+				_items = null;
```
Comment (mattleibow) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs:87]: maybe we should also do this for the dictionary?

Comment (mattleibow) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs:87]: or at least clear both lists.

Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs:87]: the `_items` is a readonly list that comes from the xplat code so I can't clear it. But I'll clear the other one

**Review by mattleibow — APPROVED** [review]

**Pull request merged**